include/grpcpp/impl/grpc_library.h

```diff
@@ -24,25 +24,34 @@
 #include <grpc/grpc.h>
 #include <grpcpp/impl/codegen/config.h>
 #include <grpcpp/impl/codegen/core_codegen.h>
-#include <grpcpp/impl/codegen/grpc_library.h>  // IWYU pragma: export
 
 namespace grpc {
 
 namespace internal {
-class GrpcLibrary final : public GrpcLibraryInterface {
+
+/// Classes that require gRPC to be initialized should inherit from this class.
+class GrpcLibrary {
  public:
-  void init() override { grpc_init(); }
-  void shutdown() override { grpc_shutdown(); }
+  explicit GrpcLibrary(bool call_grpc_init = true) : grpc_init_called_(false) {
+    if (call_grpc_init) {
+      grpc_init();
```

Comment:
It seems calling `grpc_core::initInternally()` here might experience race condition. @ctiller do you have any suggestion on this?

Reply:
@ctiller suggested use `grpc_init()` instead.